Regression tests normmix

TrN000 opened the pull request.

TrN000 committed:
version bump to next patch version

TrN000 committed:
bugfix: sigma not read correctly in no1toMmix

	covar mat will only be a vector for nor1mix models.
	have to explicitly cast to array.

TrN000 committed:
add emacs files to gitignore

TrN000 committed:
test: more class utility tests

	related to 1-dim models.

TrN000 committed:
docs: add hint on how to silence output of norMmixMLE

TrN000 committed:
feat: add sigmaAgainstModel function

	function tests that given model is at least reasonable.
	includes tests.
	namespace changes for implementation.

TrN000 committed:
feat: split expensive and cheap constructor

	now norMmix and .norMmix.
	should we not want to instantiate model with all checks, we now have that.

TrN000 committed:
test: regression test for norMmixMLE

	norMmixMLE test to ensure we handle 1-dim cases right
	(to some extent)

TrN000 committed:
change order of sigma checks

       makes it clearer when checks happen. put the checks for sigma
       next to each other.

TrN000 committed:
update readme

mmaechler commented on tests/testthat/test-norMmix.R:
```diff
@@ -1,10 +1,37 @@
   expect_equal(coerced$mu, matrix(0, 1, 2))
   expect_equal(coerced$dim, 1)
-  expect_equal(class(coerced), "norMmix")
+  expect_equal(class(coerced), "norMmix") ## FIXME: should be expect_true inherits from norMmix
```
indeed FIXME

mmaechler commented on tests/testthat/test-norMmix.R:
allgemeine Bemerkung zum  'testthat' Gebrauch:  
`expect_equal()` ist ja die testthat-obfuscation für R's eigenes  `all.equal()` und dessen `default` Methode ja  eine numerische Toleranz.
Andererseits verwenden Sie "tonnenweise"  `expect_equal(x, y)`  wo eigentlich `x == y` oder sogar `identical(x, y)` angebracht wäre (wobei zu beachten, dass `identical()`  integer und  "integer valued double" als verschieden anschaut, also, z.B.  `1 == 1L`   aber  `identical(1, 1L)`  ist falsch

TrN000 committed:
fix: use inherits to check class

Better way to do it. Doesn't break in case of weird inheritance.

TrN000 committed:
fix: vignette broken by changes in normix()

if name was null, `norMmix` would try to make one, using a variable that
wasn't defined at that point.

also my R install for 4.3.3 was somehow broken by a missing font, which
caused vignette build to fail. system update fixed one of my R installs.

TrN000 committed:
add do-* to gitignore for local scripts

TrN000 committed:
ignore .org files

TrN000 committed:
missing link in Rd file

TrN000 committed:
misc test work

	try writing a expect_identical function
	add missing error strings to test-sigmaAgainstModel

TrN000 committed:
fix: example mixture had wrong model

	should be VEI instead of EVV

TrN000 committed:
add todo about missing tests
